### ISSUE TITLE
Split minting into seperate contract

### ISSUE THREAD

**Issue by raucao** [67P/kredits-contracts-legacy#1]
In order to retain all minting history when updating the main contract, we need to split it out into a seperate, much more simple contract, which (ideally) we don't have to update afterwards, because all important mechanics live in the main contractm, which can update itself in the minting contract by way of contributor voting.

**Comment by raucao** [comment]
@bumi I think this is finished, no?

**Comment by bumi** [comment]
Yes, minting is moved into a special Token contract. But I want to review the interaction and want to simulate some contract "updates"/changes on the testnet. will close the issue once this is done.\

fyi. I am also experimenting with splitting even more from the kredits contract: like for example the contributors into a special contract

**Comment by raucao** [comment]
Cool, so this can be closed. Please open new issues for new tasks, so we're all on the same page.

> But I want to review the interaction and want to simulate some contract "updates"/changes on the testnet. will close the issue once this is done.\

Why simulate, when we have an [actual update to deploy](https://github.com/67P/kredits-contracts/pull/10) anyway?

> I am also experimenting with splitting even more from the kredits contract: like for example the contributors into a special contract

That means we currently lose all contributor data when updating the contract, correct?

**Comment by bumi** [comment]
the update is for the kredits contract - not the token contract. The token contract is written in a way that we can move balances. 

yes, currently we'd loose the contributor data - but can be scripted to be added automatically again.

**Comment by raucao** [comment]
> yes, currently we'd loose the contributor data - but can be scripted to be added automatically again.

Ok, so we need a new issue for that. Could you open one and explain what to script where?
